spline-shadow/client-ui/src/app/services/lineage/lineage-graph.service.ts

```diff
@@ -52,10 +55,12 @@ export class LineageGraphService {
     let that = this
     return that.executionPlanControllerService.lineageUsingGETResponse(executionPlanId).pipe(
       map(response => {
-        return that.convertFromExecutedLogicalPlanToFromExecutedLogicalPlanViewModel(response)
+        let convertedResponse = that.convertFromExecutedLogicalPlanToFromExecutedLogicalPlanViewModel(response)
+        this.executedLogicalPlanSubject.next(convertedResponse)
```

Comment:
I am not sure to understand. I use a pipe here.

Reply:
I meant mapping instead of piping, sorry.
From the context of the given method and its usage it seems that `executedLogicalPlanSubject` is just a stream of `activeRoute.params` flat mapped on the `executionPlanControllerService.lineageUsingGETResponse` via `toLogicalPlanView` converter function.
In other words, you take a stream of route params, then for each value you obtain a stream of lineages (or execution plans), flat them, and the resulted stream is basically your `executedLogicalPlanSubject `.
Doesn't it seem as a one liner? And so you don't even need the entire `getExecutedLogicalPlan` method?

Reply:
No Alex, the `executedLogicalPlanSubject `is not a stream of `activeRoute.params`. I've done it exactly for the same reason as `detailsInfo `: to pass the data easily to other components, but in fact this `executedLogicalPlanSubject `is useless. I just take the data of the `lineageUsingGETResponse` method and put it to another observable : `executedLogicalPlanSubject `. This one should be remove indeed. Regarding the code again there is another antipatern that I forgot the correct. The component that use the method `getExecutedLogicalPlan ` is using nested subsumption. I will commit a fix for this and remove the `executedLogicalPlanSubject `